test/extensions/common/tap/tap_matcher_test.cc

```diff
@@ -47,10 +53,43 @@ TEST_F(TapMatcherTest, Not) {
   buildMatcher(config_, matchers_);
   EXPECT_EQ(2, matchers_.size());
   statuses_.resize(matchers_.size());
-  EXPECT_FALSE(matchers_[0]->onNewStream(statuses_));
-  EXPECT_FALSE(matchers_[0]->onHttpRequestHeaders(headers_, statuses_));
-  EXPECT_FALSE(matchers_[0]->onHttpResponseHeaders(headers_, statuses_));
-  EXPECT_FALSE(matchers_[0]->matches(statuses_));
+  matchers_[0]->onNewStream(statuses_);
+  EXPECT_EQ((Matcher::MatchStatus{false, false}), matchers_[0]->matchStatus(statuses_));
+  matchers_[0]->onHttpRequestHeaders(headers_, statuses_);
+  EXPECT_EQ((Matcher::MatchStatus{false, false}), matchers_[0]->matchStatus(statuses_));
+  matchers_[0]->onHttpRequestTrailers(headers_, statuses_);
+  EXPECT_EQ((Matcher::MatchStatus{false, false}), matchers_[0]->matchStatus(statuses_));
+  matchers_[0]->onHttpResponseHeaders(headers_, statuses_);
+  EXPECT_EQ((Matcher::MatchStatus{false, false}), matchers_[0]->matchStatus(statuses_));
+  matchers_[0]->onHttpResponseTrailers(headers_, statuses_);
+  EXPECT_EQ((Matcher::MatchStatus{false, false}), matchers_[0]->matchStatus(statuses_));
+}
+
+TEST_F(TapMatcherTest, AndMightChangeStatus) {
```

Comment:
Do we need these for `Or` as well?

Reply:
I can add another test, but it's identical logic.